frontend/src/component/user/NewUser/NewUser.tsx

```diff
@@ -1,27 +1,76 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import { Box, TextField, Typography } from '@mui/material';
+import { CREATED, OK } from 'constants/statusCodes';
+import useToast from 'hooks/useToast';
 import useResetPassword from 'hooks/api/getters/useResetPassword/useResetPassword';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { useUserInvite } from 'hooks/api/getters/useUserInvite/useUserInvite';
+import { useInviteTokenApi } from 'hooks/api/actions/useInviteTokenApi/useInviteTokenApi';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import { useAuthResetPasswordApi } from 'hooks/api/actions/useAuthResetPasswordApi/useAuthResetPasswordApi';
 import AuthOptions from '../common/AuthOptions/AuthOptions';
 import DividerText from 'component/common/DividerText/DividerText';
 import { useAuthDetails } from 'hooks/api/getters/useAuth/useAuthDetails';
-import { useInviteUserToken } from 'hooks/api/getters/useInviteUserToken/useInviteUserToken';
 import ResetPasswordForm from '../common/ResetPasswordForm/ResetPasswordForm';
 import InvalidToken from '../common/InvalidToken/InvalidToken';
 import { NewUserWrapper } from './NewUserWrapper/NewUserWrapper';
+import ResetPasswordError from '../common/ResetPasswordError/ResetPasswordError';
 
 export const NewUser = () => {
     const { authDetails } = useAuthDetails();
+    const { setToastApiError } = useToast();
+    const navigate = useNavigate();
+    const [apiError, setApiError] = useState(false);
+    const [email, setEmail] = useState('');
+    const [name, setName] = useState('');
     const {
         token,
-        data,
+        data: passwordResetData,
         loading: resetLoading,
-        setLoading,
-        invalidToken,
+        isValidToken,
     } = useResetPassword();
-    const { invite, loading: inviteLoading } = useInviteUserToken();
+    const {
+        secret,
+        loading: inviteLoading,
+        isValid: isValidInvite,
+    } = useUserInvite();
+    const { addUser, loading: isUserSubmitting } = useInviteTokenApi();
+    const { resetPassword, loading: isPasswordSubmitting } =
+        useAuthResetPasswordApi();
     const passwordDisabled = authDetails?.defaultHidden === true;
 
-    if (invalidToken && !invite) {
+    const onSubmitInvitedUser = async (password: string) => {
+        try {
+            const res = await addUser(secret, { name, email, password });
+            if (res.status === CREATED) {
```

Comment:
Should we show a toast if something happens and you get something other than 200? It could happen.